### PR TITLE
Adding support for gift cards

### PR DESCRIPTION
**Pull request by bhelx**
This is adding support for the gift card features and starts the `2.4` feature branch.

## Using

```ruby
require 'securerandom'

# if we have an existing account
account_data = Recurly::Account.new(account_code: "my_account_code")

# we can use the account instance or a hash
account_data = {account_code: "my_account_code"}

# if we want to embed a new account:
account_uuid = SecureRandom.uuid
account_data = {
    account_code: account_uuid,
    email: 'sally@example.com',
    first_name: 'Sally',
    last_name: 'Smith',
    billing_info: { # can also take a token
      address1: '400 Alabama St',
      zip: '94110',
      city: 'San Francisco',
      state: 'CA',
      country: 'US',
      number: '4111-1111-1111-1111',
      month: 12,
      year: 2019
    }
}

delivery_data = {
  method: 'email',
  email_address: 'john@example.com',
  first_name: 'John',
  last_name: 'Smith',
  gifter_name: 'Sally',
  personal_message: 'Thanks John! -- sally'
}

gc = Recurly::GiftCard.new(
  product_code: 'gift_card',
  currency: 'USD',
  unit_amount_in_cents: 2000,

  # optional delivery details
  delivery: delivery_data,

  # new or existing account data
  gifter_account: account_data
)

# if you want, you can do a preview on this purchase

gc.preview
puts gc.errors #> {}

# Or you can go ahead with the purchase
gc.save!

gift_card_id = gc.id
redemption_code = gc.redemption_code

# We can find a gift card by id
gift_card = Recurly::GiftCard.find(gift_card_id)

# Or, all we need to redeem a gift card is the redemption_code
# You can build a new object from that
gift_card = Recurly::GiftCard.new(redemption_code: redemption_code)

# Subscription#create takes a gift card as a parameter
begin
  subscription = Recurly::Subscription.create(
    plan_code: 'gold',
    currency: 'USD',
    customer_notes: 'Thank you for your business!',
    account: {
      account_code: account_uuid,
    },
    gift_card: gift_card
  )
rescue Recurly::API::NotFound => e
  # will throw a not found for an invalid gift card redemption code
end

# You also have the choice to redeem the 
# card on an account directly on any given account code
gift_card.redeem(account_uuid)

puts gift_card.redeemed_at
#=> #<DateTime: 2016-07-28T21:12:58+00:00 ((2457598j,76378s,0n),+0s,2299161j)>
```

### CODE REVIEW

File: lib/recurly/api/net_http_adapter.rb [hunk 1]
```diff
@@ -60,6 +60,7 @@ def request method, uri, options = {}
           end
           http = ::Net::HTTP.new uri.host, uri.port
           http.use_ssl = uri.scheme == 'https'
+          http.verify_mode = OpenSSL::SSL::VERIFY_NONE
```
Comment (bhelx) [inline lib/recurly/api/net_http_adapter.rb]: Sorry, this will be gone.

File: lib/recurly/export_date.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module Recurly
```
Comment (bhelx) [inline lib/recurly/export_date.rb]: This got wrapped up by accident

File: lib/recurly/gift_card.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      reload response
+    rescue API::UnprocessableEntity => e
+      apply_errors e
+    end
+
+    private
+
+    def xml_keys
+      keys = super
+      keys << 'redemption_code' if redemption_code? && !redemption_code_changed?
```
Comment (bhelx) [inline lib/recurly/gift_card.rb]: This is kind of a hack to ensure that `redemption_code` is always rendered in xml. Stole this from how `Account` handles `account_code`.

**Comment by csmb** [comment]
+1

**Pull request merged**